builtin/logical/pki/chain_util.go

```diff
@@ -417,15 +425,29 @@ func (sc *storageContext) rebuildIssuersChains(referenceCert *issuerEntry /* opt
 	}
 
 	// Finally, write all issuers to disk.
+	//
+	// See the note above when sorting issuers for why we delay persisting
+	// the referenceCert, if it was provided.
 	for _, issuer := range issuers {
 		entry := issuerIdEntryMap[issuer]
 
+		if referenceCert != nil && issuer == referenceCert.ID {
+			continue
+		}
+
 		err := sc.writeIssuer(entry)
 		if err != nil {
 			pretty := prettyIssuer(issuerIdEntryMap, issuer)
 			return fmt.Errorf("failed to persist issuer (%v) chain to disk: %v", pretty, err)
 		}
 	}
+	if referenceCert != nil {
+		err := sc.writeIssuer(issuerIdEntryMap[referenceCert.ID])
+		if err != nil {
+			pretty := prettyIssuer(issuerIdEntryMap, referenceCert.ID)
+			return fmt.Errorf("failed to persist issuer (%v) chain to disk: %v", pretty, err)
```

Comment:
nit: Missed it earlier, but this might be better with the %w although probably not going to change much in the end. If you do address it, there is the one above as well that has the same thing.

```suggestion
			return fmt.Errorf("failed to persist issuer (%v) chain to disk: %w", pretty, err)
```

Reply:
Thanks! I've updated it throughout the file, though other locations probably need updating as well.